tests/actions.go

```diff
@@ -811,3 +851,473 @@ func checkoutTag(tagName string) error {
+	if err != nil {
+		return fmt.Errorf("failed to launch scheduler backup job: %v, %s", err, string(res))
+	}
+	return nil
+}
+
+//restore genereted record by scheduledbackup, and make sure the data is correct
+func (oa *operatorActions) CheckScheduledBackup(info *TidbClusterInfo) error {
+	glog.Infof("begin to check scheduler backup")
+	defer func() {
+		glog.Infof("deploy check scheduler end")
+	}()
+
+	jobName := fmt.Sprintf("%s--scheduled-backup",info.ClusterName)
+	fn := func() (bool, error) {
+		job, err :=  oa.kubeCli.BatchV1beta1().CronJobs(info.Namespace).Get(jobName,metav1.GetOptions{})
+		if err != nil {
+			glog.Errorf("failed to get cronjobs %s ,%v",jobName,err)
+			return false, nil
+		}
+
+		jobs, err := oa.kubeCli.BatchV1().Jobs(info.Namespace).List(metav1.ListOptions{})
+		if err != nil {
+			glog.Errorf("failed to list jobs %s ,%v",info.Namespace,err)
+			return false, nil
+		}
+
+		backupJobs := []batchv1.Job{}
+		for _,j := range jobs.Items {
+			if pid, found := getParentUIDFromJob(j); found && pid == job.UID {
+				backupJobs = append(backupJobs,j)
+			}
+		}
+
+		if len(backupJobs) == 0 {
+			glog.Errorf("cluster [%s] scheduler jobs is creating, please wait!",info.ClusterName)
+			return false, nil
+		}
+
+		for _,j := range backupJobs {
+			if j.Status.Succeeded == 0 {
+				glog.Error("cluster [%s] back up job is not completed, please wait! ",info.ClusterName);
+				return false, nil
+			}
+		}
+
+		return true, nil
+	}
+
+	err := wait.Poll(WAITSECOND * time.Second, WAITMINUTE * time.Minute, fn)
+	if err != nil {
+		return fmt.Errorf("failed to launch scheduler backup job: %v", err)
+	}
+
+	return nil
+}
+
+//full data backup
+func (oa *operatorActions) DeployAdHocBackup(info *TidbClusterInfo) error {
+	glog.Infof("begin to deploy adhoc backup")
+	defer func() {
+		glog.Infof("deploy adhoc backup end")
+	}()
+	sets := map[string]string {
+		"clusterName":		info.ClusterName,
+		"name":				"test-backup",
+		"mode":				"backup",
+		"user":				"root",
+		"password":			info.Password,
+	}
+	var buffer bytes.Buffer
+	for k,v := range sets {
+		set := fmt.Sprintf(" --set %s=%s",k , v)
+		_,err := buffer.WriteString(set)
+		if err != nil  {
+			return err
+		}
+	}
+
+	setStr := buffer.String()
+	fullbackupName := fmt.Sprintf("%s-backup",info.ClusterName)
+	cmd := fmt.Sprintf("helm install -n %s --namespace %s /charts/%s/tidb-backup %s",
+		fullbackupName, info.Namespace, info.OperatorTag, setStr)
+	glog.Infof("install adhoc deployment [%s]",cmd)
+	res, err := exec.Command("/bin/sh", "-c", cmd).CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("failed to launch adhoc backup job: %v, %s", err, string(res))
+	}
+	return nil
+}
+
+//restore full data backup is correct and check full data backup data is correct
+func (oa *operatorActions) CheckAdHocBackup(info *TidbClusterInfo) error {
+	glog.Infof("begin to clean adhoc backup")
+	defer func() {
+		glog.Infof("deploy clean backup end")
```

Comment:
clean backup -> check backup

Reply:
clean is not necessary， for the CleanTidbCluster function will delete all the resources。 but check is necessary for wait the back completed